File: 226-invert-binary-tree/solution.cpp
// Runtime: 0ms, Beats 100.00% of users with Python3
// Memory: 11.39MB, Beats 51.44% of users with Python3

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr) return nullptr;
        invertTree(root->left);
        invertTree(root->right);
        TreeNode* temp = root->right;
        root->right = root->left;
        root->left = temp;
        return root;
    }
};
        

File: 226-invert-binary-tree/solution.py
# Runtime: 38ms, Beats 37.48% of users with Python3
# Memory: 16.57MB, Beats 34.33% of users with Python3

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return
        root.right, root.left = root.left, root.right
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root
        